src/Immutable/Collection.php

```diff
@@ -224,17 +224,26 @@ public function reject(callable $func): ImmutableList
      */
     public function any(callable $func): bool
     {
-        return $this->filter($func)->getSize() >= 1 ? true : false;
+        $list   = $this->getList();
+        $size   = $list->count();
+        $result = false;
+
+        for ($idx = 0; $idx < $size; $idx += 1) {
+            if ($func($list[$idx])) {
+                $result = true;
+                break;
+            }
+        }
+
+        return $result;
     }
 
     /**
      * {@inheritdoc}
      */
     public function every(callable $func): bool
     {
-        $init = $this->getSize();
-
-        return $this->filter($func)->getSize() == $init ? true : false;
+        return $this->reject($func)->getSize() == 0;
```

Comment:
How about using strict equal (`===`) to check result is `0`?

It will be faster than using loose equal (`==`) operator.

Reply:
Do you have proof of this?

Reply:
It seems that do either strict or loose equal comparison is identical for performance.

Because of other reasons like readability, and `$this->reject($func)->getSize()` result value, we should be very clear to know the result value will be the same as compared value and type.

Unless the `$this->reject($func)->getSize()` result will not always return the same type.

Reply:
Also, what about edge cases like comparisons of objects? I feel that `==` is apt.

Reply:
I think the better question to ask is, how significantly impacted is algorithm performance? Are the gains from change far more significant than the costs associated with not making the change?

Reply:
Maybe this issue will be at the next PR because this issue I think it's not related to this PR directly.

I need to write other test cases and ensure that the `$this->reject($func)->getSize()`  result type is always `int`.

Reply:
All right. You're welcome to test as much as you need to.